Set is_non_overlapping_and_dense_ flag in OpaqueTensorImpl constructor

SS-JIA opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49470 Set is_non_overlapping_and_dense_ flag in OpaqueTensorImpl constructor**

https://github.com/pytorch/pytorch/pull/48625 changes the default contiguous settings for `TensorImpl` causing the Vulkan backend to crash. Therefore, add argument that can set `is_non_overlapping_and_dense_` back to false for `OpaqueTensorImpl` constructor.

Differential Revision: [D25592826](https://our.internmc.facebook.com/intern/diff/D25592826)

Commit:
Add is_non_overlapping_and_dense argument for OpaqueTensorImpl

[ghstack-poisoned]

Commit:
Update on "Add is_non_overlapping_and_dense argument for OpaqueTensorImpl"




[ghstack-poisoned]

AshkanAliabadi commented:
Not sure who the right person is to review this patch.  The backstory is that we ran into runtime crashes on the Vulkan backend recently which Stephen has narrowed to #48625.

ngimel commented:
Can you please add tests that would exercise codepaths previously leading to crashes? We are working on making TensorImpl more efficient, and unless you are testing for the behaviors you rely upon, it could lead to breakages in the future. 
As an alternative to `refresh_contiguous()`, which is relatively expensive, you could correctly set contiguity bitfields in constructor, similarly to how it has been done for SparseTensorImpl constructor in #48625

Commit:
Update on "Call refresh_contiguous() in OpaqueTensorImpl constructor"


https://github.com/pytorch/pytorch/pull/48625 changes the default contiguous settings for TensorImpl. Therefore, refresh_contiguous() should be called in OpaqueTensorImpl to get the correct settings.

[ghstack-poisoned]

Commit:
Update on "Call refresh_contiguous() in OpaqueTensorImpl constructor"


https://github.com/pytorch/pytorch/pull/48625 changes the default contiguous settings for TensorImpl. Therefore, refresh_contiguous() should be called in OpaqueTensorImpl to get the correct settings.

[ghstack-poisoned]

SS-JIA commented:
As suggested, changed to just set the flag in the constructor.

@xta0 may want to take a look at this as well, since OpaqueTensorImpl is also used by the Metal backend.

AshkanAliabadi commented:
Yeah we are puzzled by why this was not caught by CI as well.  Investigating.  Thanks.

Commit:
Update on "Set is_non_overlapping_and_dense_ flag in OpaqueTensorImpl constructor"


https://github.com/pytorch/pytorch/pull/48625 changes the default contiguous settings for `TensorImpl` causing the Vulkan backend to crash. Therefore, set `is_non_overlapping_and_dense_` back to false for `OpaqueTensorImpl`.

[ghstack-poisoned]
